linkerd/cache/src/layer.rs

```diff
@@ -0,0 +1,72 @@
+use crate::Service;
+use futures::Future;
+use linkerd2_stack::NewService;
+use std::time::Duration;
+use tracing::info_span;
+use tracing_futures::Instrument;
+
+#[derive(Clone, Debug)]
+struct Config {
+    capacity: usize,
+    max_idle_age: Duration,
+}
+
+/// A layer that that builds a routing service.
+///
+/// A `Rec`-typed `Recognize` instance is used to produce a target for each
+/// `Req`-typed request. If the router doesn't already have a `Service` for this
+/// target, it uses a `Mk`-typed `Service` stack.
```

Comment:
nit: this looks like the same comment the old router layer...maybe:
```suggestion
/// A layer that that builds a cache service.
```
(and perhaps add some notes on the cache's eviction policy or something?)